flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/JobReconciler.java

```diff
@@ -114,6 +119,9 @@ public void reconcile(FlinkDeployment flinkApp, Context context, Configuration e
     }
 
     private boolean readyForSpecChanges(FlinkDeployment deployment) {
+        if (isUpgradeModeChangedToLastState(deployment)) {
+            return isJobRunning(deployment);
```

Comment:
I think we could allow upgrading to last-state even SUSPENDED.  So we could simply change the IF branch at  line 125 to:

```
if (deployment.getSpec().getJob().getUpgradeMode() != UpgradeMode.SAVEPOINT && !isUpgradeModeChangedToLastState(deployment))
```

This way if previous job was with savepoint we would pick up the last savepoint  otherwise we either have state or not depending on the HA setting. Which would be fair for a suspended job.

Reply:
I think you are right. We should allow upgrade to last-state even SUSPENDED.